### DISCUSSION TITLE
Retire Settings of Back Links (they are actually Outgoing Links) to Book and Chapter Discussion

### DISCUSSION THREAD

**Discussion by tim-hub** [tim-hub/obsidian-bible-reference#115] [Ideas]
![image](https://github.com/tim-hub/obsidian-bible-reference/assets/2884320/f195837a-e2ed-4679-af3b-03fa0694e4de)

Hi All, I am thinking of retiring these 2 settings in next big release 1.9.x, I feel like the UI for these 2 settings are not very good, and it is required users to fill the whole book or the whole chapter to get it making sense. 

![image](https://github.com/tim-hub/obsidian-bible-reference/assets/2884320/a1ed7c4e-6e6c-4e52-9a6a-be3accaa6a89)

This is a discussion thread, please let me know your thoughts, :)

**Comment by tim-hub** [comment]
reasons why I prefer removing it for now. 

https://github.com/tim-hub/obsidian-bible-reference/pull/119#issuecomment-1705014684

**Comment by minermaniac447** [comment]
Looks like github dropped my original reply, whoops.

My thoughts on this echo what I expressed in the collapsible verses setting PR, but I feel that removing settings or customization, especially if they were there in the first place, is almost never the best option. I also would argue that this fits in best with Obsidian's principles as a whole - provide as much customizeability to the user and as much control, both as they want, and as possible, as well as making portability a priority.

While I understand the concern of having many settings, I don't think that should be a reason to stop providing the settings or not make them, any more than increasing code complexity is a reason to stop developing on a project. It shouldn't be done without consideration, but this plugin, even if we had two to three times the number of settings, would be far from any majorly adopted plugin - most of which have many, many more settings than ours. I do think some of these will naturally go away as well, or be combined together with other settings to reduce the overall complexity of the page (something like #109, for example, will probably make the setting that this discussion is about obsolete, as well as many tag settings). We can also mitigate the growing number of settings with better UI organization - something I'd really like to implement is collapsible sections of related settings.

Touching back on customizability and portability - while I recognize and personally greatly prefer the use of tags, backlinks are by far the best option for anyone using an internal bible - and that internal bible is much more portable (i.e. if Obsidian becomes a crypto scam tomorrow, the internal bible and everything set up surrounding it will still work just fine on another platform with no changes needed, while tagging may not work as nicely). If it were a question of do we add it as an option originally, I would be less aggressively for it, but given that it's something we already had, I'm greatly against taking it away (which is my mentality for features in general).

TL;DR I think leaving in the setting/s, even if it's a bit messy and may become obsolete, is best both for backwards compatibility purposes, as well as leaning on Obsidian's principles of portability and customizability.

**Comment by iamsangpark** [comment]
I recently updated the plugin and was really bummed to find out these options are gone. I recently spend chunk of time importing bible texts into Obsidian and renaming all the files so that it matches the backlink'd name from this plugin... 

I think the main reason that I would prefer using links over tags is the graph view. Some one from my church showed me this and this is what convinced me to fully switch over to Obsidian for taking notes. With the links, I can go to any book & chapter of the bible and immediately see all the notes I have taken that references that particular chapter. And it's really easy to navigate between the notes and the chapters and see different connections and whatnot. It's possible to see the tags in the graph view, but I don't think it displays the back-tags(?) so you are forced to search for the tags which to me isn't as intuitive and easy to use. 

I agree with @minermaniac447 that the settings didn't seem all that complicated... it was pretty straightforward and made total sense.

I would love for this feature to come back and I hope that you will consider bringing it back. Meanwhile, I'm resorting to manually adding the links each time I use the plugin to insert a passage

**Reply by tim-hub** [reply]
Thanks for the feedback.

Will try to bring it back.
